### PR TITLE
Feat: Storybook: Improve component organisation - Navigation Category - Issue #66275 

### PR DESCRIPTION
**Pull request by hbhalodia**
## What?

- Issue - #66275 
- Changes the componenet metadata to use the sitemap structure to ease user navigation to storybook.

## Why?
This change is part of the larger [Storybook Improvements](https://make.wordpress.org/design/2024/09/17/design-systems-storybook-improvements/) and the [shared Sitemap](https://www.figma.com/board/0ZfHa37xxWoMqy7McezUNX/WordPress.org-Storybook-Sitemap?node-id=126-5118&t=ruCtnVTDVjHTfksc-1). 

## How?

Adding Categories that helps organize and group components into sections based on their role and function within the library makes our library easier to navigate and understand. 

## Testing Instructions

1. Checkout to the PR.
2. Run `npm run storybook:dev`.
3. Check for the updated components with category that are changed in PR.

### Testing Instructions for Keyboard

- None

## Screenshots or screencast 
Screenshot 2024-11-01 at 9 25 55 AM
Screenshot 2024-11-01 at 9 26 00 AM

### CODE REVIEW

File: packages/components/src/tabs/stories/index.story.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ import Tooltip from '../../tooltip';
 import Icon from '../../icon';
 
 const meta: Meta< typeof Tabs > = {
-	title: 'Components/Containers/Tabs',
+	title: 'Components/Navigation/Tabs',
```
Comment (tyxla) [inline packages/components/src/tabs/stories/index.story.tsx]: I think we should revert this change. See @mirka's [comment](https://github.com/WordPress/gutenberg/pull/66072#pullrequestreview-2365228539) about it, as to why `Tabs` and `TabPanel` make most sense in "Containers". 

Let's confirm with @WordPress/gutenberg-design, and in case Figma's outdated, it can be good to update it.

File: packages/components/src/tree-grid/stories/index.story.tsx [hunk 2]
```diff
@@ -16,7 +16,8 @@ import { Button } from '../../button';
 import InputControl from '../../input-control';
 
 const meta: Meta< typeof TreeGrid > = {
-	title: 'Components (Experimental)/TreeGrid',
+	title: 'Components (Experimental)/Navigation/TreeGrid',
```
Comment (tyxla) [inline packages/components/src/tree-grid/stories/index.story.tsx:19]: Another thing (cc @ciampo @mirka): experimental components will fall in their own section down below, meaning they will have their own Navigation directory. Is that desired? Should we consider marking the experimental components just with badges, like `CustomSelectControlV2` is marked WIP?

File: packages/components/src/tree-grid/stories/index.story.tsx [hunk 3]
```diff
@@ -16,7 +16,8 @@ import { Button } from '../../button';
...
+	id: 'components-treegrid',
```
Comment (tyxla) [inline packages/components/src/tree-grid/stories/index.story.tsx]: As this component is still formally experimental, if we change its ID, it will cause the previous URL not to work:

http://localhost:50240/?path=/docs/components-experimental-treegrid--docs

To fix that, we have 2 options:
- Use `id: 'components-experimental-treegrid'`
- Add the component to `PREVIOUSLY_EXPERIMENTAL_COMPONENTS`. 

Since the component is still experimental, I'd prefer the former option. Once we stabilize the component, we can always add it to `PREVIOUSLY_EXPERIMENTAL_COMPONENTS`

**Review by tyxla — COMMENTED** [review]
Thanks for your work here @hbhalodia! 🙌

Just like in #66657 there are a few considerations for experimental components we might need to consider before shipping.

Also, I'd like to double-check if we really want to put "Tabs" into the Navigation category. My expectation is that it's better suited to live in Containers instead.

File: packages/components/src/tabs/stories/index.story.tsx (hunk 1 above)
Comment (hbhalodia) [reply packages/components/src/tabs/stories/index.story.tsx]: Thanks @tyxla, I would revert this change to use tabs and tabpanel in `Containers` category.

File: packages/components/src/tree-grid/stories/index.story.tsx (hunk 2 above)
Comment (hbhalodia) [reply packages/components/src/tree-grid/stories/index.story.tsx:19]: > experimental components will fall in their own section down below, meaning they will have their own Navigation directory.

Waiting on this until confirmation of above.

Cc: @tyxla

Comment (tyxla) [reply packages/components/src/tree-grid/stories/index.story.tsx:19]: I've been discussing this with @ciampo, and we concluded that it's a problem that can be solved separately. 

In the future we may remove the "Experimental" distinction from the Storybook component list, and those categories that we're introducing will be on the top level. That being said, we could leave things as they are and ship this PR, and I've opened #66805 to discuss this separately.

**Review by tyxla — APPROVED** [review]
LGTM, thanks again @hbhalodia 🙌
